sdk/webpubsub/Microsoft.Azure.WebJobs.Extensions.WebPubSub/src/Bindings/WebPubSubRequestValueProvider.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host.Bindings;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.WebJobs.Extensions.WebPubSub
+{
+    internal class WebPubSubRequestValueProvider : IValueProvider
+    {
+        private readonly WebPubSubRequest _request;
+
+        public Type Type { get; }
+
+        public WebPubSubRequestValueProvider(WebPubSubRequest request, Type type)
+        {
+            _request = request;
+            Type = type;
+        }
+
+        public Task<object> GetValueAsync()
+        {
+            return Task.FromResult(GetRequest());
+        }
+
+        public string ToInvokeString()
+        {
+            return nameof(WebPubSubRequest);
+        }
+
+        private object GetRequest()
+        {
+            if (Type == typeof(JObject))
+            {
+                return JObject.FromObject(_request);
+            }
+            if (Type == typeof(string))
+            {
+                return JObject.FromObject(_request).ToString();
```

Comment:
>_request [](http://example.com/codeflow?start=42&length=8)

What's this case?

Reply:
As a common support that if customer want to bind request to string type.